lib/redis/event_handler.js

```diff
@@ -39,6 +39,25 @@ exports.connectHandler = function (self) {
 
     self.initParser();
 
+    self.lastWriteTime = null;
+    if (typeof self.options.timeoutPerCommand === 'number') {
+      var timeoutPerCommand = self.options.timeoutPerCommand;
+      debug('Per-command timeout set to %s', self.options.timeoutPerCommand);
+      self.timeoutPerCommand = setInterval(() => {
```

Comment:
how about using this: https://nodejs.org/api/net.html#net_socket_settimeout_timeout_callback to generate a timeout event from the socket if it becomes inactive while the queue is being processed?

Reply:
This timeout guards against a socket which has no activity.  The value to use probably greatly depends on the use cases.  In a production environment, there's probably enough traffic to keep such a timeout from triggering.  But in a dev environment, there might not be any traffic for relatively long stretches of time making this an idle connection which then times out.  Setting a per command timeout seems better.

Also, if this socket timeout occurs, it doesn't enable specific error handling / recovery for the specific commands which timed out.

Or were you suggesting this as an additional check and not an alternative solution?

Reply:
I am suggesting to set this to something reasonable while we are processing commands and to disable it while the socket is not used.
It just seems that it can be easier to implement than setting and cancelling timeouts in the lib. And more powerful as well.